### PR TITLE
Log node info on test failures

### PR DESCRIPTION
**Pull request by brendandburns**
In order to help debug problems with the disruptive e2e tests.

### CODE REVIEW

**Comment by brendandburns** [comment]
@ihmccreery @ixdy @quinton-hoole

**Comment by ixdy** [comment]
@brendandburns did you intend to pull in test/e2e/reboot.go?

**Comment by ixdy** [comment]
This is also now failing to build.

File: test/e2e/util.go [hunk 1]
```diff
@@ -1465,8 +1465,27 @@ func dumpAllPodInfo(c *client.Client) {
 	logPodStates(pods.Items)
 }
 
+func dumpAllNodeInfo(c *client.Client) {
+	nodes, err := c.Nodes().List(labels.Everything(), fields.Everything())
+	if err != nil {
+		Logf("unable to fetch node list: %v", err)
```
Comment (ixdy) [inline test/e2e/util.go]: maybe return after this too?

Comment (ixdy) [inline test/e2e/util.go]: or maybe not. but does `List()` return [] or nil on error?

Comment (brendandburns) [inline test/e2e/util.go]: I agree with the original comment.  return added.

**Comment by brendandburns** [comment]
ptal.  reboot.go is  on purpose so that we get that extra info.

**Comment by ixdy** [comment]
lgtm

@k8s-bot e2e test this please

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Comment by ixdy** [comment]
@k8s-bot unit test this

**Pull request merged**